### ISSUE TITLE
[cuckoo] ERROR: Cuckoo is already running. PID: 1691

### ISSUE THREAD

**Issue by nice9uy** [cuckoosandbox/cuckoo#2358]
when i run cuckoo, i got some error like this

(cuckoo-env) root@cuckoo:~# cuckoo 

  _____________________________________/\/\_______________________________
  ___/\/\/\/\__/\/\__/\/\____/\/\/\/\__/\/\__/\/\____/\/\/\______/\/\/\___
  _/\/\________/\/\__/\/\__/\/\________/\/\/\/\____/\/\__/\/\__/\/\__/\/\_
  _/\/\________/\/\__/\/\__/\/\________/\/\/\/\____/\/\__/\/\__/\/\__/\/\_
  ___/\/\/\/\____/\/\/\/\____/\/\/\/\__/\/\__/\/\____/\/\/\______/\/\/\___
  ________________________________________________________________________

 Cuckoo Sandbox 2.0.6
 www.cuckoosandbox.org
 Copyright (c) 2010-2018

2018-07-06 15:21:00,894 [cuckoo] ERROR: Cuckoo is already running. PID: 1691
(cuckoo-env) root@cuckoo:~# 

and then when i running server with cuckoo web runserver, i got error like this

(cuckoo-env) root@cuckoo:~# cuckoo web runserver 172.24.8.84:8000
Performing system checks...

System check identified no issues (0 silenced).
July 06, 2018 - 15:23:14
Django version 1.8.4, using settings 'cuckoo.web.web.settings'
Starting development server at http://172.24.8.84:8000/
Quit the server with CONTROL-C.
[06/Jul/2018 15:23:21] "GET /dashboard/ HTTP/1.1" 200 22290
[06/Jul/2018 15:23:21] "GET /static/css/main.css HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/css/vendor.css HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/vendor.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/handlebars-templates.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/hexdump.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/loader.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/sticky.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/favicon-32x32.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/analysis_sidebar.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/analysis_feedback.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/submission.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/process_tree.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/recent.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/app.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/rdp.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/js/cuckoo/analysis_network.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/graphic/cuckoo_inverse.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/graphic/cuckoo-coffee-cup.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/images/prev.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/images/next.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/images/loading.gif HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "GET /static/images/close.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:21] "POST /analysis/api/tasks/recent/ HTTP/1.1" 200 13
2018-07-06 15:23:21,679 [django.request] ERROR: Internal Server Error: /cuckoo/api/status
Traceback (most recent call last):
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/cuckoo/web/utils.py", line 55, in inner
    return func(*args, **kwargs)
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/django/views/decorators/http.py", line 45, in inner
    return func(request, *args, **kwargs)
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/cuckoo/web/controllers/cuckoo/api.py", line 45, in status
    temp_file = Files.temp_put("")
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/cuckoo/common/files.py", line 97, in temp_put
    prefix="upload_", dir=path or temppath()
  File "/usr/lib/python2.7/tempfile.py", line 314, in mkstemp
    return _mkstemp_inner(dir, prefix, suffix, flags)
  File "/usr/lib/python2.7/tempfile.py", line 244, in _mkstemp_inner
    fd = _os.open(file, flags, 0600)
OSError: [Errno 2] No such file or directory: '/tmp/cuckoo-tmp-root/upload_BWWRmP'
[06/Jul/2018 15:23:21] "GET /cuckoo/api/status HTTP/1.1" 500 12976
[06/Jul/2018 15:23:25] "GET /dashboard/ HTTP/1.1" 200 22290
[06/Jul/2018 15:23:25] "GET /static/css/vendor.css HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/css/main.css HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/vendor.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/handlebars-templates.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/hexdump.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/loader.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/sticky.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/analysis_sidebar.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/favicon-32x32.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/analysis_feedback.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/submission.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/process_tree.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/recent.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/analysis_network.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/rdp.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/js/cuckoo/app.js HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/graphic/cuckoo_inverse.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/graphic/cuckoo-coffee-cup.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/images/prev.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/images/next.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/images/loading.gif HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "GET /static/images/close.png HTTP/1.1" 304 0
[06/Jul/2018 15:23:25] "POST /analysis/api/tasks/recent/ HTTP/1.1" 200 13
2018-07-06 15:23:25,880 [django.request] ERROR: Internal Server Error: /cuckoo/api/status
Traceback (most recent call last):
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/cuckoo/web/utils.py", line 55, in inner
    return func(*args, **kwargs)
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/django/views/decorators/http.py", line 45, in inner
    return func(request, *args, **kwargs)
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/cuckoo/web/controllers/cuckoo/api.py", line 45, in status
    temp_file = Files.temp_put("")
  File "/root/cuckoo-env/local/lib/python2.7/site-packages/cuckoo/common/files.py", line 97, in temp_put
    prefix="upload_", dir=path or temppath()
  File "/usr/lib/python2.7/tempfile.py", line 314, in mkstemp
    return _mkstemp_inner(dir, prefix, suffix, flags)
  File "/usr/lib/python2.7/tempfile.py", line 244, in _mkstemp_inner
    fd = _os.open(file, flags, 0600)
OSError: [Errno 2] No such file or directory: '/tmp/cuckoo-tmp-root/upload_J7vFHg'
[06/Jul/2018 15:23:25] "GET /cuckoo/api/status HTTP/1.1" 500 12976

So, How to fix it??

Thanks

**Comment by RicoVZ** [comment]
Hi nice9uy,

Thanks for posting an issue.

There seem to be two seperate issues here. The 'already running' error occurs if a previously started Cuckoo instance from the same working directory is still running. You can verify what the process is by using `ps -p <pid>`, and kill it if you want. After this, Cuckoo should start.

About the 'no such file ..' error, I suspect the `/tmp/cuckoo-tmp-root` directory does not exist. In this case, Cuckoo should first be started once under the user you are running web on. When Cuckoo starts, the Cuckoo temp dir for that user is created.

Tldr:
1. Kill the running Cuckoo
2. Start Cuckoo under the user you want to run Cuckoo web on so that the temp dir is created.
3. Start web.

Note: we do not recommend running Cuckoo or any Cuckoo apps as root (except for Cuckoo rooter). This especially goes for web development servers in debug mode.

**Comment by nice9uy** [comment]
It's works thanks
